apps/src/lib/script-editor/LessonGroupCard.jsx

```diff
@@ -0,0 +1,139 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import ReactDOM from 'react-dom';
+import {connect} from 'react-redux';
+import color from '../../util/color';
+import {borderRadius} from './constants';
+import OrderControls from './OrderControls';
+import LessonCard from './LessonCard';
+import {addLesson, removeGroup, moveGroup} from './editorRedux';
+
+const styles = {
+  groupHeader: {
+    fontSize: 18,
+    color: 'white',
+    background: color.cyan,
+    borderTopLeftRadius: borderRadius,
+    borderTopRightRadius: borderRadius,
+    padding: 10
+  },
+  groupBody: {
+    background: color.lightest_cyan,
+    borderBottomLeftRadius: borderRadius,
+    borderBottomRightRadius: borderRadius,
+    padding: 10,
+    marginBottom: 20
+  },
+  addLesson: {
+    fontSize: 14,
+    color: '#5b6770',
+    background: 'white',
+    border: '1px solid #ccc',
+    boxShadow: 'none',
+    margin: '0 10px 10px 10px'
+  }
+};
+
+class LessonGroupCard extends Component {
+  static propTypes = {
+    lessonGroup: PropTypes.object.isRequired,
+    lessonGroupCount: PropTypes.number.isRequired,
+    setLessonMetrics: PropTypes.func.isRequired,
+    setTargetLesson: PropTypes.func.isRequired,
+    targetLessonPos: PropTypes.number,
+    lessonMetrics: PropTypes.object.isRequired,
+
+    //redux
+    addLesson: PropTypes.func.isRequired,
+    moveGroup: PropTypes.func.isRequired,
+    removeGroup: PropTypes.func.isRequired
+  };
+
+  handleAddLesson = lessonGroupPosition => {
+    const newLessonName = prompt('Enter new lesson name');
+    if (newLessonName) {
+      this.props.addLesson(lessonGroupPosition, newLessonName);
+    }
+  };
+
+  handleMoveLessonGroup = direction => {
+    if (
+      (this.props.lessonGroup.position !== 1 && direction === 'up') ||
+      (this.props.lessonGroup.position !== this.props.lessonGroupCount &&
+        direction === 'down')
+    ) {
+      this.props.moveGroup(this.props.lessonGroup.position, direction);
+    }
+  };
+
+  handleRemoveLessonGroup = () => {
```

Comment:
It was a bit hard to see what you changed here, because the extraction of LessonGroupCard is in the same commit as the extraction of this method. next time you are moving a large chunk of code without changing it, please put that in its own commit. in this case, that would have meant extracting LessonGroupCard, without also extracting handleRemoveLessonGroup . In addition to making this easier to review, you may find that it's easier to develop, too, since it lets you get your code into a working state and makes it easier to see a smaller diff of what you've changed as you go to extract handleRemoveLessonGroup.

Reply:
Thanks for the suggestion Dave!